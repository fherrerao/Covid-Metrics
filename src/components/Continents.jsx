import { useSelector } from "react-redux";
import { NavLink } from "react-router-dom";

const Continents = () => {
  const arrContinents = useSelector((state) => state.metricsReducer);
  console.log(arrContinents);

  return (
    <ul>
      {arrContinents.map((country) => (
        <NavLink to={`/countries/${country.country}`} key={country.id}>
          <div>{country.country}</div>
        </NavLink>
      ))}
    </ul>
  );
};
export default Continents;
