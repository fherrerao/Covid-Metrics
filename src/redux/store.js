import { combineReducers, compose, applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import metricsReducer from './metrics/metrics';

const reducer = combineReducers({
  metricsReducer,
})

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducer,
  composeEnhancers(applyMiddleware(thunk, logger)),
);

export default store;
