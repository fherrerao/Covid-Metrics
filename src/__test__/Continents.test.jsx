import React from "react";
import renderer from "react-test-renderer";
import store from "../redux/store";
import { Provider } from "react-redux";
import Continents from "../components/Continents";

describe("Testing Continents", () => {
  it("Render the continents, create a snapshot", () => {
    const list = renderer
      .create(
        <Provider store={store}>
          <Continents />
        </Provider>
      )
      .toJSON();
    expect(list).toMatchSnapshot();
  });
});
