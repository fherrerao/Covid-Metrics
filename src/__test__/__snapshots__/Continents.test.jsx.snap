// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Continents Render the continents, create a snapshot 1`] = `
<div
  className="bg-first"
>
  <div
    className="d-flex justify-content-center align-items-center py-5 gap-5 bg-second"
  >
    <img
      alt=""
      src={Array []}
      width="200px"
    />
    <p
      className="f-gill fs"
    />
  </div>
  <ul
    className="g-countries p-0"
  />
</div>
`;
